py/objnamedtuple.c

```diff
@@ -109,7 +109,9 @@ STATIC mp_obj_t namedtuple_make_new(const mp_obj_type_t *type_in, size_t n_args,
     }
 
     // Create a tuple and set the type to this namedtuple
-    mp_obj_tuple_t *tuple = MP_OBJ_TO_PTR(mp_obj_new_tuple(num_fields, NULL));
+    // Explicit allocation; mp_obj_new_tuple with num_fields=0 returns read-only object
+    mp_obj_tuple_t *tuple = mp_obj_malloc_var(mp_obj_tuple_t, mp_obj_t, num_fields, &mp_type_tuple);
+    tuple->len = num_fields;
     tuple->base.type = type_in;
```

Comment:
I think we could pass `type_in` to `mp_obj_malloc_var()` directly and delete this line to override the type.

Reply:
Yes that would be good!

Reply:
Great suggestion, changed that in d2da4b4